pkg/loop/service.go

```diff
@@ -49,12 +49,12 @@ type pluginService[P grpcPlugin, S services.Service] struct {
 	newService func(context.Context, any) (S, error)
 
 	serviceCh chan struct{} // closed when service is available
-	service   S
+	Service   S
 
-	testInterrupt chan func(*pluginService[P, S]) // tests only (via TestHook) to enable access to internals without racing
+	testInterrupt chan func(*PluginService[P, S]) // tests only (via TestHook) to enable access to internals without racing
 }
 
-func (s *pluginService[P, S]) init(pluginName string, p P, newService func(context.Context, any) (S, error), lggr logger.Logger, cmd func() *exec.Cmd, stopCh chan struct{}) {
+func (s *PluginService[P, S]) Init(pluginName string, p P, newService func(context.Context, any) (S, error), lggr logger.Logger, cmd func() *exec.Cmd, stopCh chan struct{}) {
```

Comment:
Why is this exported?

Reply:
It's used by the `reporting_plugins` package where the generic services are located.